### ISSUE TITLE
StandardMessageCodec  Unsupported value: true

### ISSUE THREAD

**Issue by BigTimo** [flutter/flutter#65619]
log:
java.lang.IllegalArgumentException:Unsupported value: true
io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:294)

see
https://github.com/flutter/engine/blob/b3e281fc0c6e7f2b1abb0bb68f1fee3545ce8339/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L211

value == Boolean.TRUE or value == Boolean.FALSE  always be false

change 
```java
protected void writeValue(ByteArrayOutputStream stream, Object value) {
    if (value == null || value.equals(null)) {
      stream.write(NULL);
     } else if (value == Boolean.TRUE) {
      stream.write(TRUE);
     } else if (value == Boolean.FALSE) {
      stream.write(FALSE);
     } else if (value instanceof Number) {
     ...
    }
}
```
to

```java
protected void writeValue(ByteArrayOutputStream stream, Object value) {
    if (value == null || value.equals(null)) {
      stream.write(NULL);
    } else if (value instanceof Boolean) {
      stream.write(Boolean.TRUE.equals(value) ? TRUE : FALSE);
    } else if (value instanceof Number) {
     ...
    }
}
```

**Comment by darshankawar** [comment]
Hi @BigTimo ,
Can you please elaborate on when / how  do you get this error ? Can you provide a minimal complete reproducible code sample along with `flutter doctor -v` ?
Thanks.

**Comment by a651828650** [comment]
> Hi @BigTimo ,
> Can you please elaborate on when / how do you get this error ? Can you provide a minimal complete reproducible code sample along with `flutter doctor -v` ?
> Thanks.

1.17.5

**Comment by allenymt** [comment]
This bug has not yet been  merged into 1.22, I have check the branch origin/flutter-1.22-candidate.13 and origin/flutter-1.22-candidate.12

**Comment by allenymt** [comment]
I guess it is caused by Serializable，This bug has not yet been merged into 1.22

So temporarily solve it like this，
Traverse map replacement Boolean 
`
 try{
            // https://github.com/flutter/flutter/issues/65619
            if (args!=null && args instanceof Map) {
                Map<Object, Object> replaceBooleanMap = (Map<Object, Object>) args;
                if (!replaceBooleanMap.isEmpty()){
                    Iterator<Map.Entry<Object, Object>> replaceBooleanIterator = replaceBooleanMap.entrySet().iterator();
                    while (replaceBooleanIterator.hasNext()) {
                        Map.Entry<Object, Object> entry = replaceBooleanIterator.next();
                        if (entry.getValue() instanceof Boolean) {
                            Boolean b = (Boolean) entry.getValue();
                            if (b.equals(Boolean.TRUE)) {
                                replaceBooleanMap.put(entry.getKey(), Boolean.TRUE);
                            } else {
                                replaceBooleanMap.put(entry.getKey(), Boolean.FALSE);
                            }
                        }
                    }
                }
            }
        } catch (Exception e){
            e.printStackTrace();
        }

`
